### PR TITLE
MULE-18894: NotSerializableException when persisting ABP decorators.

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: integration/src/test/java/org/mule/test/components/PayloadDecoratorThroughQueuesTestCase.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    this.payload = payload;
+  }
+
+  @Override
+  protected String getConfigFile() {
+    return "org/mule/test/components/payload-through-vm-config.xml";
+  }
+
+  @Test
+  public void testPayload() throws Exception {
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/components/PayloadDecoratorThroughQueuesTestCase.java:53]: allure

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: integration/src/test/java/org/mule/test/streaming/PayloadStatisticsTestCase.java [hunk 2]
```diff
@@ -134,6 +135,33 @@ public void bytesSource() throws MuleException, IOException, TimeoutException {
...
+
+    final PayloadStatistics fileListStatistics =
+        muleContext.getStatistics().getPayloadStatistics("bytesSourceThroughVM/source");
+
+    assertThat(fileListStatistics.getComponentIdentifier(), is("http:listener"));
+
+    assertThat(fileListStatistics.getInvocationCount(), is(1L));
+
+    assertThat(fileListStatistics.getInputObjectCount(), is(0L));
+    assertThat(fileListStatistics.getInputByteCount(), is(0L));
```
Comment (elrodro83) [inline integration/src/test/java/org/mule/test/streaming/PayloadStatisticsTestCase.java:160]: assert the input of the vm publish as well

File: integration/src/test/resources/org/mule/streaming/payload-statistics-non-repeatable-config.xml [hunk 3]
```diff
@@ -98,4 +118,10 @@
             <test-components:queue-push config-ref="queue" consumeStream="true"/>
         </foreach>
     </sub-flow>
+    
+    <flow name="consumeStreamThroughVM">
+        <vm:listener queueName="Q1" config-ref="VM_Config"/>
```
Comment (elrodro83) [inline integration/src/test/resources/org/mule/streaming/payload-statistics-non-repeatable-config.xml:123]: add the non-repeatable config

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: integration/src/test/resources/org/mule/streaming/payload-statistics-non-repeatable-config.xml (hunk 3 above)
Comment (fsgonz) [reply integration/src/test/resources/org/mule/streaming/payload-statistics-non-repeatable-config.xml:123]: vm does not support streaming.

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**